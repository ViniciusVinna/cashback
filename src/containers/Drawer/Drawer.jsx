import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Heading from 'components/Heading';
import Waves from 'components/Waves';

import { DrawerStyled } from './Drawer.styled';

const {
  BackIcon,
  Content,
  DrawerWrapper,
  Header,
  HeaderContainer,
  Overlay,
} = DrawerStyled;

const Drawer = ({ children, extraPadding, isVisible, onClose, title }) => (
  <Fragment>
    <DrawerWrapper isVisible={isVisible} data-testid="drawer">
      <Header>
        <HeaderContainer>
          <BackIcon data-testid="close-drawer" onClick={onClose} />
          <Heading level="h4">{title}</Heading>
        </HeaderContainer>
        <Waves />
      </Header>

      <Content extraPadding={extraPadding}>
        {children}
      </Content>
    </DrawerWrapper>

    <Overlay
      onClick={onClose}
      isVisible={isVisible}
    />
  </Fragment>
);

Drawer.propTypes = {
  children: PropTypes.node.isRequired,
  extraPadding: PropTypes.bool,
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
};

Drawer.defaultProps = {
  extraPadding: false,
};

export default Drawer;
