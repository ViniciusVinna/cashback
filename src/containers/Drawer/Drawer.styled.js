import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { IoMdArrowBack } from 'react-icons/io';

export const DrawerStyled = {
  BackIcon: styled(IoMdArrowBack)(({ theme }) => css`
    color: ${theme.color.n100};
    cursor: pointer;
    font-size: ${theme.iconSize.m};
  `),
  DrawerWrapper: styled('div')(({ theme, isOpen }) => css`
    backface-visibility: hidden;
    background-color: ${theme.color.n200};
    box-shadow: ${theme.boxShadow.modal};
    display: flex;
    flex-direction: column;
    height: 100%;
    overflow: auto;
    position: fixed;
    right: 0;
    top: 0;
    transform: ${isOpen ? 'translateX(0)' : 'translateX(100%)'};
    transition: ${theme.transition.bezier};
    width: 100%;
    z-index: ${theme.zIndex.drawer};
  `),
  Header: styled('header')(({ theme }) => css`
    top: 0;
    width: 100%;
    margin-bottom: -${theme.space.xl};
  `),
  HeaderContainer: styled('header')(({ theme }) => css`
    align-items: center;
    background-color: ${theme.color.g500};
    display: flex;
    justify-content: flex-start;
    padding: ${theme.space.xs} ${theme.space.m} 0;
    top: 0;
    width: 100%;

    & h4 {
      color: ${theme.color.n100};
      font-size: ${theme.textSize.l};
      margin: auto;
    }
  `),
  Content: styled('div')(({ theme, extraPadding }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    padding: ${theme.space.xs} ${extraPadding ? theme.space.xl : theme.space.m} ${theme.space.m};
    width: 100%;

    & p {
      font-size: ${theme.textSize.m};
      color: ${theme.color.d400};
      margin-bottom: ${theme.space.xl};
    }
  `),
  Overlay: styled('div')(({ theme, isOpen }) => css`
    backface-visibility: hidden;
    background-color: ${theme.color.shadow};
    bottom: 0;
    left: 0;
    opacity: ${isOpen ? 0.4 : 0};
    overflow: hidden;
    position: fixed;
    right: 0;
    top: 0;
    transition: ${theme.transition.bezier};
    visibility: ${isOpen ? 'visible' : 'hidden'};
    z-index: ${theme.zIndex.overlay};
  `),
};
