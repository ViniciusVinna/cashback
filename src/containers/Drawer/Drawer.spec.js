import React from 'react';
import Drawer from './Drawer';

const mockDispatch = jest.fn();

const props = {
  children: <div>child</div>,
  isDrawerActive: true,
  onClose: mockDispatch,
  title: 'Drawer Title',
};

const setup = (ownProps = props) => {
  const {
    getByTestId,
    getByText,
    queryByTestId,
  } = render(<Drawer {...ownProps} />);

  const container = getByTestId('drawer');

  return {
    container,
    getByTestId,
    getByText,
    queryByTestId,
  };
};

describe('Drawer', () => {
  describe('basic functionality', () => {
    it('it should be a render children', () => {
      const { container } = setup();
      expect(container).toMatchSnapshot();
    });
  });
});
