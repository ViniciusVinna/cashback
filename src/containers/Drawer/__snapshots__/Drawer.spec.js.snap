// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer it should render properly 1`] = `
<div>
  <div
    class="css-lvnw3k"
    data-testid="drawer"
  >
    <header
      class="css-k5577l"
    >
      <header
        class="css-16zn5ys"
      >
        <svg
          class="css-d4tpuw"
          data-testid="close-drawer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
          />
        </svg>
        <h4
          class="css-tzbvbu"
        >
          Drawer Title
        </h4>
      </header>
      <svg
        viewBox="0 0 1440 320"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0,32L60,58.7C120,85,240,139,360,144C480,149,600,107,720,80C840,53,960,43,1080,74.7C1200,107,1320,181,1380,218.7L1440,256L1440,0L1380,0C1320,0,1200,0,1080,0C960,0,840,0,720,0C600,0,480,0,360,0C240,0,120,0,60,0L0,0Z"
          fill="#52b947"
          fill-opacity="1"
        />
      </svg>
    </header>
    <div
      class="css-1tc17gr"
    >
      <div>
        child
      </div>
    </div>
  </div>
  <div
    class="css-6h85ol"
  />
</div>
`;
